packages/url_launcher/ios/Classes/UrlLauncherPlugin.m

```diff
@@ -118,11 +127,15 @@ - (void)launchURL:(NSString *)urlString result:(FlutterResult)result {
 
 - (void)launchURLInVC:(NSString *)urlString result:(FlutterResult)result {
   NSURL *url = [NSURL URLWithString:urlString];
-
-  SFSafariViewController *safari = [[SFSafariViewController alloc] initWithURL:url];
   _currentSession = [[FLTUrlLaunchSession alloc] initWithUrl:url withFlutterResult:result];
-  safari.delegate = _currentSession;
-  [_viewController presentViewController:safari animated:YES completion:nil];
+  [_viewController presentViewController:_currentSession.safari animated:YES completion:nil];
+}
+
+- (void)closeWebView:(NSString *)urlString result:(FlutterResult)result {
+  if (_currentSession != nil) {
+    [_currentSession close];
+  }
```

Comment:
Please double check that the completion handler is called in this path. If not, _currentSession should be deallocated here as well.